src/state/opamFileTools.ml

```diff
@@ -1192,6 +1192,10 @@ let read_opam dir =
       (OpamPp.string_of_bad_format (OpamPp.Bad_format (snd err)));
     None
   | None, None -> None
+  | exception (OpamPp.Bad_version _bf) ->
+    log "opam-version unsupported on %s. Add as dummy unavailable package."
+      (OpamFile.to_string opam_file);
+    Some (OpamFile.OPAM.with_available (FBool false) OpamFile.OPAM.empty)
```

Comment:
I would tweak this slightly: we have the version above, so given:

```opam
opam-version: "3.0"
some-field-that-does-not-exist: true
```
we effectively translate it to:
```opam
opam-version: "2.0"
available: opam-version >= "3.0"
synopsis: "Requires opam 3.0"
description: """
This package uses opam 3.0 file format which opam 2.2 cannot read.

In order to install or view information on this package, please upgrade
your opam installation to at least version 3.0."""
```
This has the benefit that if the user attempts to install it, they get a better error message about the lack of availability, and if they do `opam show` or `opam list`, then they get some description about why the package is missing.